model/task/task.go

```diff
@@ -2572,35 +2581,90 @@ func ArchiveMany(tasks []Task) error {
+				bson.A{ // Pipeline
+					bson.D{{Key: "$set", Value: bson.D{ // execution = execution + 1
+						{Key: ExecutionKey, Value: bson.D{
+							{Key: "$add", Value: bson.A{
+								"$" + ExecutionKey, 1,
+							}},
+						}},
+					}}},
+					bson.D{{Key: "$unset", Value: bson.A{
+						AbortedKey,
+						AbortInfoKey,
+						OverrideDependenciesKey,
+					}}}})
+			if err != nil {
+				return nil, errors.Wrap(err, "archiving tasks")
+			}
 		}
+		if len(execTasksIds) > 0 {
+			// Backwards compatibility call + LPE setting for all tasks
+			_, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
+				bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: execTasksIds}}}}, // Query all execution tasks
+				bson.A{ // Pipeline
+					bson.D{{Key: "$set", Value: bson.D{ // Sets LatestParentExecution (LPE) = !exists(LPE) ? execution + 1 : LPE + 1
+						{Key: LatestParentExecutionKey, Value: bson.D{
+							{Key: "$cond", Value: bson.A{
+								bson.D{{Key: "$not", Value: bson.A{ // !exists(LPE)
+									"$" + LatestParentExecutionKey,
+								}}},
+								bson.D{{Key: "$add", Value: bson.A{ // execution + 1
+									"$" + ExecutionKey, 1,
+								}}},
+								bson.D{{Key: "$add", Value: bson.A{ // LPE + 1
+									"$" + LatestParentExecutionKey, 1,
+								}}},
+							}}},
+						}}},
+					}})
+
+			// Same call without backwards compatibility
+			// _, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
+			// 	bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: toUpdateExecTaskIds}}}}, // Query all 'taskIds'
+			// 	bson.A{ // Pipeline
+			// 		bson.D{{Key: "$set", Value: bson.D{ // Sets LatestParentExecution (LPE) = LPE + 1
+			// 			{Key: LatestParentExecutionKey, Value: bson.D{{Key: "$add", Value: bson.A{
+			// 				"$" + LatestParentExecutionKey, 1,
+			// 			}}}}}},
+			// 		}})
 
-		taskColl := evergreen.GetEnvironment().DB().Collection(Collection)
-		_, err = taskColl.UpdateMany(ctx, bson.M{
-			IdKey: bson.M{
-				"$in": taskIds,
-			},
-		},
-			bson.M{
-				"$unset": bson.M{
-					AbortedKey:   "",
-					AbortInfoKey: "",
-				},
-				"$inc": bson.M{
-					ExecutionKey: 1,
-				},
-			})
-		if err != nil {
-			return nil, err
+			if err != nil {
+				return nil, errors.Wrap(err, "Error updating documents")
+			}
+
+			// Call to update all tasks that are actually restarting
+			_, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
+				bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: toUpdateExecTaskIds}}}}, // Query all archiving execution tasks
+				bson.A{ // Pipeline
+					bson.D{{Key: "$set", Value: bson.D{ // Execution = LPE
+						{Key: ExecutionKey, Value: "$" + LatestParentExecutionKey}}},
+					},
+					bson.D{{Key: "$unset", Value: bson.A{
+						AbortedKey,
+						AbortInfoKey,
+						OverrideDependenciesKey,
+					}}}})
+
+			return nil, errors.Wrap(err, "Error updating documents")
```

Comment:
"Updating documents" to follow error convention

Reply:
Updated! I also updated other areas where it started with error